.github/workflows/reusable-pr-is-ready.yaml

```diff
@@ -0,0 +1,91 @@
+name: PR Complete
+
+on:
+  workflow_call:
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  All_checkboxes_checked:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Fetch PR body and validate checkboxes
+        run: |
+          # 1. Fetch the pull request body via GitHub API
+          PR_BODY=$(curl -s \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }} \
+          | jq -r .body)
+
+          echo "Pull Request Body:"
+          echo "$PR_BODY"
+
+          # 2. Look for any unchecked boxes ("[ ]")
+          UNCHECKED=$(echo "$PR_BODY" | grep '\[ \]' || true)
+
+          # 3. If any are found, fail the build
+          if [ -n "$UNCHECKED" ]; then
+            echo "Error: Found unchecked checkboxes:"
+            echo "$UNCHECKED"
+            exit 1
+          else
+            echo "All checkboxes appear to be checked!"
+          fi
+  Docs_added_or_not_needed:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install yq (for YAML parsing)
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y yq
+    
+      - name: Validate feature files and doc changes
+        run: |
+          set -euo pipefail
+    
+          KEYS=("feature" "enhancement" "deprecation")
+          CHANGED_FILES=$(git diff --name-only "${{ github.event.pull_request.base.sha }}" HEAD)
+          CHANGED_FEATURE_FILES=$(echo "${CHANGED_FILES}" | grep '^docs/source/releases/latest/.*\.yaml$' || true)
+          CHANGED_DOC_FILES=$(echo "${CHANGED_FILES}" | grep '^docs/.*\.rst$' || true)
```

Comment:
Can you add comments to these two lines to make it more obvious what they're doing?

Reply:
Done